core/src/main/java/io/questdb/cutlass/http/client/HttpClient.java

```diff
@@ -75,35 +78,48 @@ public void close() {
             Unsafe.free(bufLo, bufferSize, MemoryTag.NATIVE_DEFAULT);
             bufLo = 0;
         }
-        responseHeaders = Misc.free(responseHeaders);
+        responseHeaders.free();
     }
 
     public void disconnect() {
-        socket.close();
+        socket = Misc.free(socket);
     }
 
-    public Request newRequest() {
+    public Request newRequest(CharSequence host, int port) {
+        if (socket == null || socket.isClosed()) {
+            this.socket = socketFactory.newInstance(nf, LOG);
+            this.host = host;
+            this.port = port;
+        } else {
+            // Connection reuse.
+            if (!Chars.equals(this.host, host)) {
+                throw new HttpClientException("same host expected for connection reuse [host=").put(this.host).put(']');
+            }
+            if (this.port != port) {
+                throw new HttpClientException("same host expected for connection reuse [port=").put(this.port).put(']');
```

Comment:
```suggestion
                throw new HttpClientException("same port expected for connection reuse [port=").put(this.port).put(']');
```

Reply:
Thanks for spotting the typo. Fixed in 5f1b452ef9cafe595241a4015d7596c97746600d